geometric_kernels/kernels/matern_kernel.py

```diff
@@ -119,12 +122,18 @@ def __new__(
 
         elif isinstance(space, NoncompactSymmetricSpace):
             num = num or default_num(space)
-            key = kwargs.get("key", B.create_random_state())
-            feature_map = default_feature_map(
-                space,
-                kernel=kernel,
-                num=num,
-            )
+            if "key" in kwargs:
+                key = kwargs["key"]
+            else:
+                raise Exception(
```

Comment:
Better to raise a specific Exception, like `ValueError`

Reply:
Done